File: src/test/java/se/johannesdahlgren/aoc24/Day12Test.java
package se.johannesdahlgren.aoc24;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.Test;

public class Day12Test {

  @Test
  void part1() {
    long count = new Day12().solve();
    assertThat(count).isEqualTo(1402544L);
  }

}

File: src/test/kotlin/se/johannesdahlgren/aoc22/Day12Test.kt
package se.johannesdahlgren.aoc22

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class Day12Test {
    @Test
    fun part1Example() {
        val day12 = Day12("day12Example")
        assertThat(day12.fewestStepToBestSignal()).isEqualTo(31)
    }

    @Test
    fun part1() {
        val day12 = Day12("day12")
        assertThat(day12.fewestStepToBestSignal()).isEqualTo(408)
    }

    @Test
    fun part2Example() {
        val day12 = Day12("day12Example")
        assertThat(day12.fewestStepToBestSignalStartAny()).isEqualTo(29)
    }

    @Test
    fun part2() {
        val day12 = Day12("day12")
        assertThat(day12.fewestStepToBestSignalStartAny()).isEqualTo(399)
    }
}